src/logger.cpp

```diff
@@ -85,7 +85,11 @@ size_t Logger::errorCount() const
 
 ErrorPtr Logger::getError(size_t index) const
 {
-    return mPimpl->mErrors.at(index);
+    ErrorPtr err = nullptr;
+    if (index < mPimpl->mErrors.size()) {
+        err = mPimpl->mErrors.at(index);
+    }
+    return err;
```

Comment:
Hmm, I guess it's a matter of taste. Actually, now that you pointed out to me that `index` is an `unsigned int`, I would probably have:

```
ErrorPtr Logger::getError(size_t index) const
{
    return (index < mPimpl->mErrors.size()) ? mPimpl->mErrors.at(index) : nullptr;
}
```

Anyway, I am not sure it's worth fighting over.